cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java

```diff
@@ -0,0 +1,268 @@
+            Assert.fail("bigQueryData is null");
+            return result;
+        }
+        // Get the column count of the first JsonObject in bigQueryData
+        int jsonObjectIdx = 0;
+        int columnCountTarget = 0;
+        if (bigQueryData.size() > 0) {
+            columnCountTarget = bigQueryData.get(jsonObjectIdx).entrySet().size();
+        }
+        // Compare the number of columns in the source and target
+        Assert.assertEquals("Number of columns in source and target are not equal",
+                columnCountSource, columnCountTarget);
+
+        while (rsSource.next()) {
+            int currentColumnCount = 1;
+            while (currentColumnCount <= columnCountSource) {
+                String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
+                int columnType = mdSource.getColumnType(currentColumnCount);
+                String columnName = mdSource.getColumnName(currentColumnCount);
+                // Perform different comparisons based on column type
+                switch (columnType) {
+                    case Types.BIT:
+                        boolean bqDateString = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBoolean();
+                        result = getBooleanValidation(rsSource, String.valueOf(bqDateString), columnName, columnTypeName);
+                        Assert.assertTrue("Different values found for column : %s", result);
+                        break;
+
+                    case Types.DECIMAL:
+                    case Types.NUMERIC:
+                        BigDecimal sourceDecimal = rsSource.getBigDecimal(currentColumnCount);
+                        BigDecimal targetDecimal = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBigDecimal();
+                        int desiredScale = 2; // Set the desired scale (number of decimal places)
+                        BigDecimal adjustedSourceValue = sourceDecimal.setScale(desiredScale, RoundingMode.HALF_UP);
+                        BigDecimal adjustedTargetValue = targetDecimal.setScale(desiredScale, RoundingMode.HALF_UP);
+                        Decimal sourceDecimalValue = Decimal.fromDecimal(adjustedSourceValue);
+                        Decimal targetDecimalValue = Decimal.fromDecimal(adjustedTargetValue);
+                        Assert.assertEquals("Different values found for column : %s", sourceDecimalValue, targetDecimalValue);
+                        break;
+
+                    case Types.REAL:
+                        float sourceReal = rsSource.getFloat(currentColumnCount);
+                        float targetReal = bigQueryData.get(jsonObjectIdx).get(columnName).getAsFloat();
+                        Assert.assertTrue(String.format("Different values found for column : %s", columnName),
+                                Float.compare(sourceReal, targetReal) == 0);
+                        break;
+
+                    case Types.TIMESTAMP:
+                        break;
```

Comment:
As discussed we need to add the logic here we are not going to simply add break here.

Reply:
done